libnetwork/internal/l2disco/unsol_arp_linux.go

```diff
@@ -0,0 +1,79 @@
+package l2disco
+
+import (
+	"encoding/binary"
+	"fmt"
+	"net"
+	"slices"
+
+	"golang.org/x/sys/unix"
+)
+
+var (
+	arpTemplate = []byte{
+		0x00, 0x01, // Hardware type
+		0x08, 0x00, // Protocol
+		0x06,       // Hardware address length
+		0x04,       // IPv4 address length
+		0x00, 0x01, // ARP request
+		0x00, 0x00, 0x00, 0x00, 0x00, 0x00, // Sender MAC
+		0x00, 0x00, 0x00, 0x00, // Sender IP
+		0x00, 0x00, 0x00, 0x00, 0x00, 0x00, // Target MAC (always zeros)
+		0x00, 0x00, 0x00, 0x00, // Target IP
+	}
+	bcastMAC = []byte{0xff, 0xff, 0xff, 0xff, 0xff, 0xff}
+)
+
+type UnsolARP struct {
+	pkt []byte
+	sd  int
+	sa  *unix.SockaddrLinklayer
+}
+
+// NewUnsolARP returns a pointer to an object that can send unsolicited ARPs on
+// the interface with ifIndex, for ip and mac.
+func NewUnsolARP(ip net.IP, mac net.HardwareAddr, ifIndex int) (*UnsolARP, error) {
+	sd, err := unix.Socket(unix.AF_PACKET, unix.SOCK_DGRAM|unix.SOCK_CLOEXEC, int(htons(unix.ETH_P_ARP)))
```

Comment:
I considered using github.com/mdlayher/socket, but [it should not be used directly in end user applications](https://github.com/mdlayher/socket#:~:text=This%20package%20should%20not%20be%20used%20directly%20in%20end%20user%20applications.).

A system under so much pressure that it can't squeeze out an ARP packet for any significant period of time has bigger issues than this thread.

(github.com/mdlayher/arp has no tagged releases, and no updates for a couple of years. I didn't look any closer than that.)